src/inspector_agent.cc

```diff
@@ -429,9 +431,38 @@ class ChannelImpl final : public v8_inspector::V8Inspector::Channel {
   std::unique_ptr<v8_inspector::V8InspectorSession> session_;
 };
 
+class RepeatingInspectorTimer {
+ public:
+  RepeatingInspectorTimer(uv_loop_t* loop,
+                          double interval_s,
+                          V8InspectorClient::TimerCallback callback,
+                          void* data) : timer_(),
+                                        callback_(callback),
+                                        data_(data) {
+    uv_timer_init(loop, &timer_);
+    int64_t interval_ms = 1000 * interval_s;
+    uv_timer_start(&timer_, OnTimer, interval_ms, interval_ms);
+  }
+
+  RepeatingInspectorTimer(const RepeatingInspectorTimer&) = delete;
+
+  ~RepeatingInspectorTimer() {
+    uv_timer_stop(&timer_);
```

Comment:
This should uv_close() the timer (and not free the memory until the close callback is called.)